blaze-core/src/main/scala/org/http4s/blaze/Http1Stage.scala

```diff
@@ -234,22 +238,23 @@ object Http1Stage {
     else CachedEmptyBufferThunk
   }
 
-  /** Encodes the headers into the Writer, except the Transfer-Encoding header which may be returned
+  /** Encodes the headers into the Writer. Does not encode `Transfer-Encoding` or
+    * `Content-Length` headers, which are left for the body encoder. Adds
+    * `Date` header if one is missing and this is a server response.
+    *
     * Note: this method is very niche but useful for both server and client. */
-  def encodeHeaders(headers: Headers, rr: Writer, isServer: Boolean): Option[`Transfer-Encoding`] = {
-    var encoding: Option[`Transfer-Encoding`] = None
+  def encodeHeaders(headers: Iterable[Header], rr: Writer, isServer: Boolean) = {
```

Comment:
There are already several traversals of the List since in  Http1ServerStage.scala we also get the `Connection` and `Content-Length` headers. So if optimization in this is important then it should be optimized for those two also. No point doing it here just for one header of three.